app/api/dao/user.py

```diff
@@ -231,7 +231,7 @@ def update_user_profile(user_id: int, data: Dict[str, str]):
             user_with_same_username = UserModel.find_by_username(username)
 
             # username should be unique
-            if user_with_same_username:
+            if user_with_same_username and user_with_same_username.id != user_id:
```

Comment:
catch integrity error instead of doing this?

Reply:
@epicadk I made the changes.

Reply:
Then check for this after catching Integrity Error :thinking: 

What do you think @epicadk

Reply:
In that case, what would be the point of catching Integrity Error?

Reply:
> Then check for this after catching Integrity Error 🤔
> 
> What do you think @epicadk

Yup, what @devkapilbansal said. IS there not a particular code for an Integrity error where the unique condition is not met?